packages/core/src/components/forms/_file-upload.scss

```diff
@@ -63,6 +63,10 @@ $file-upload-button-padding-large: ($pt-input-height-large - $pt-button-height)
   &.pt-fill {
     width: 100%;
   }
+
+  &.pt-large {
```

Comment:
What about the `.pt-large &` case? See [L103](https://github.com/palantir/blueprint/pull/1586/files#diff-fa25bb68536fc9097a2548cf91d320e1R103).

Reply:
@cmslewis not sure that's relevant actually... this line applies to a different element than L103.

Reply:
Right, but when the `pt-file-upload` element is made to be large via cascading styles (by applying `pt-large` to a parent), then `&.pt-large` styles won't increase this element's height in that case. Should it?

I linked the L103 as an example of the selector I think we should use if so.

Reply:
I just tested it and it looks like we should @cmslewis (cascaded `.pt-large` changes the visual size but not the actual size):
<img width="407" alt="screen shot 2017-09-20 at 3 41 33 pm" src="https://user-images.githubusercontent.com/199754/30671129-38e6d564-9e1a-11e7-87d0-9c9ade145ba0.png">

Reply:
@Landish - per @llorca's investigation above, can you change the selector to `.pt-large &`?

Reply:
@cmslewis 

So, you're saying to just change selector and everything should work? 

That selector compiles to this code:

<img width="366" alt="screen shot 2017-09-21 at 14 05 36" src="https://user-images.githubusercontent.com/534610/30690663-ffdc6420-9ed5-11e7-8f13-fcf7751cc6be.png">

Just confirm it and I'll update the PR.

Reply:
Yep, that's correct, reasoning being that `pt-file-upload` is one of the several components that can receive Large styling in two ways: 

- by adding `pt-large` directly to it (`&.pt-large`)
- by adding `pt-large` to an ancestor (`.pt-large &`)

We should fix the height in both cases.